main.cc

```diff
@@ -1422,7 +1422,11 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 startlog.warn("Using default cluster name is not recommended. Using a unique cluster name will reduce the chance of adding nodes to the wrong cluster by mistake");
             }
             auto group0_id = sys_ks.local().get_raft_group0_id().get();
-            auto gossiper_seeds = get_seeds_from_db_config(*cfg, broadcast_addr);
+
+            // Fail on a gossiper seeds lookup error only if the node is not bootstrapped.
+            const bool fail_on_lookup_error = !sys_ks.local().bootstrap_complete();
+
+            auto gossiper_seeds = get_seeds_from_db_config(*cfg, broadcast_addr, fail_on_lookup_error);
```

Comment:
If seeds parameter is indeed unused after bootstrap is completed may be we can just skip the call to `get_seeds_from_db_config` completely?

Reply:
Seed nodes are still used after bootstrap -- gossiper tries to gossip with them.

There is one very rare edge case where this is useful: full cluster shutdown, followed by IP change for every node. Seeds are the only way to re-establish connectivity between nodes in this scenario, by pointing to the new IP addresses

Reply:
> Seed nodes are still used after bootstrap -- gossiper tries to gossip with them.

What if all the nodes in the seed list are replaced?

> 
> There is one very rare edge case where this is useful: full cluster shutdown, followed by IP change for every node. Seeds are the only way to re-establish connectivity between nodes in this scenario, by pointing to the new IP addresses

Indeed. So seeds are used after bootstrap completes after all. So why treat errors differently if this is the case? What if we not throw on bootstrap as well? In the best case a bootsrapping node will use less seeds, in the worst case there will be zero seeds and the bootstrap will fail.

Reply:
> What if all the nodes in the seed list are replaced?

Nothing bad will happen, gossip will just fail to reach out to them but otherwise everything will function.

> In the best case a bootsrapping node will use less seeds, in the worst case there will be zero seeds and the bootstrap will fail.

This makes it easier for cluster admin to shoot themselves in the foot, and (after recent experiences in the filed) I'm 100% sure they will if we give them the opportunity.

So they will for example bootstrap 2 separate clusters, when they wanted 1, and then call support and support will call us. (Also note that zero seeds doesn't mean that bootstrap will fail -- it actually means the node will become a new cluster)

Reply:
> Also note that zero seeds doesn't mean that bootstrap will fail -- it actually means the node will become a new cluster

We should probably make it fail regardless. It will be less error prone. But until this is the case we need to fail boostrap here indeed.